### ISSUE TITLE
google_container_cluster - network and subnetwork diff not being suppressed

### ISSUE THREAD

**Issue by mrparkers** [terraform-providers/terraform-provider-google#2202]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

Terraform: v0.11.8
Provider: v1.9.0

### Affected Resource(s)

- `google_container_cluster`

### Terraform Configuration Files

```tf
resource "google_container_cluster" "dev" {
    project                  = "${google_project.dev.project_id}"
    name                     = "dev"
    min_master_version       = "1.10.7-gke.2"
    region                   = "us-central1"
    private_cluster          = true
    master_ipv4_cidr_block   = "172.28.80.0/28"
    additional_zones         = [
        "us-central1-a",
        "us-central1-b",
        "us-central1-c",
        "us-central1-f"
    ]
    network_policy {
        enabled = true
    }
    network                  = "${var.vpc_shared_network_self_link}"
    subnetwork               = "${var.k8s_subnetwork_self_link}"
    ip_allocation_policy {
        cluster_secondary_range_name  = "pods"
        services_secondary_range_name = "services"
    }
    lifecycle {
        ignore_changes = [
            "node_pool"
        ]
    }
    node_pool {
        name = "default-pool"
    }
    remove_default_node_pool = true
    depends_on               = [
        "google_project_service.compute",
        "google_project_service.container"
    ]
}
```

### Expected Behavior

All `tf apply` and `tf plan` runs should show an empty diff

### Actual Behavior

All `tf apply` and `tf plan` runs show the following diff:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.dev.google_container_cluster.dev
      network:    "projects/my-project/global/networks/shared-network" => "https://www.googleapis.com/compute/v1/projects/my-project/global/networks/shared-network"
      subnetwork: "projects/my-project/regions/us-central1/subnetworks/dev-k8s" => "https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/subnetworks/dev-k8s"


Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. Create a network and subnet
2. Use the computed `self_link` properties to fill in the `network` and `subnet` fields for google_container_cluster
3. `tf apply`
4. `tf plan`

### Important Factoids

My guess is that a simple `DiffSuppressFunc` will fix this, since all runs of `tf apply` are instant and don't result in any changes to my GKE cluster.

I'll submit a PR for this if that is all that's required, assuming I did not mess something up in my config.

**Comment by danawillow** [comment]
Hey @mrparkers, this is a long-standing bug with TF core, unfortunately: https://github.com/hashicorp/terraform/issues/18209. In the meantime, my suggestion would be to use a separate node_pool resource so you don't have to do lifecycle.ignore_changes. Going to go ahead and close this out; feel free to follow along with the core issue if you'd like!

**Comment by mrparkers** [comment]
Thanks for the quick response!

I am using a separate `node_pool` resource, the lifecycle code is there to help with removing the default node pool when spinning the cluster up.

I ended up just building the `projects/my-project/global/networks/shared-network` string using variables from my network config instead of using the `self_link`.

Also, I did my part and 👍ed the issue in tf core :smile:

Thanks again!
